softwareComponents/configuration/include/configuration/joints.hpp

```diff
@@ -49,11 +51,58 @@ struct Joint: public atoms::VisitableBase< Joint, JointVisitor > {
         return _positions;
     }
 
+    /**
+     * @brief Sets joint settings to corresponding values given by <pos>.
+     * Assumes there are exactly as many values as there are joint settings.
+     * 
+     * @throws std::logic_error if the given settings do not respect joint limits.
+     */
     void setPositions( std::span< const float > pos ) {
-        assert( pos.size() == _positions.size() && "Positions have to preserve given size" );
+        assert( pos.size() == _positions.size() && "Insufficient number of parameters to set" );
+
+        size_t pId = 0;
+        for ( auto[low, high] : _jointLimits )
+        {
+            if ( pos[pId] < low || high < pos[pId] ) 
+                throw std::logic_error( 
+                    fmt::format( "Parameter at index {} with value {} is not within limit [{}, {}]", 
+                    pId, pos[pId], low, high ) );
+            ++pId;
+        }
+
         std::copy( pos.begin(), pos.end(), _positions.begin() );
     }
 
+    /**
+     * @brief Changes each joint setting by the corresponding value in <diff>.
+     * 
+     * Assumes there are exactly as many values as there are joint settings.
+     * 
+     * @return result error if the resulting values do not respect joint limits.
+     */
+    atoms::Result< std::monostate > changePositionsBy( std::span< float > diff )
+    {
+        assert( diff.size() == _positions.size() && "Insufficient number of parameters to set" );
+
+        // Add up current parameter values with diff
+        std::transform( diff.begin(), diff.end(), _positions.begin(), diff.begin(),
+            []( float current, float difference ){ return current + difference; } );
```

Comment:
Toto má nepřijemnou vlastnost, že to mění vstupní argument (což bych u funkce nečekal). Proto funkce `setPositions` bere `const float`.

Řešení jsou dvě:
- prvně provést kontrolu a poté zapsat
- konstruovat nový vektor, zkontrolovat a poté udělat swap.